src/components/Dialog.vue

```diff
@@ -53,10 +53,7 @@
 <script lang="ts">
 import Vue from 'vue'
 import axios from '@nextcloud/axios'
-import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
-import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
-import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
-import NcPasswordField from '@nextcloud/vue/dist/Components/NcPasswordField.js'
+import { NcButton, NcModal, NcNoteCard, NcPasswordField } from '@nextcloud/vue'
```

Comment:
This is not a good idea as long as we use webpack for applications!

Because you set the dependency to external webpack now has to treeshake and it is quite bad doing it.
In numbers, a simple app just consisting of the example code in the readme:

With this: `9543055` Bytes ~ 9,1 MiB
When keeping the default import: `2674259` Bytes ~ 2,55 MiB

Reply:
Thanks for that insight @susnux I think this is really bad for app that use more components and we should rather consider moving everything to importing the default export of the lib instead of the dist folder. As far as my tests went the dist imports are not tree-shakable at all.

There is probably no optimal way (except from moving away from web pack to site), but the `index.module.js` of `@nextcloud/vue` is currently at 883K which seems to be a value that a lot of apps and also server easily exceed with the individual modules from dist.

Just out of curiosity, du you have the small test app in some repo? I'd be curious to see the actual impact in the bundles with `webpack-bundle-analyzer` or `vite-bundle-visualizer`

Reply:
Seems like a good idea to let vite do the tree-shaking here then?

Reply:
Thanks for that insight @susnux I think this is really bad for app that use more components and we should rather consider moving everything to importing the default export of the lib instead of the dist folder. As far as my tests went the dist imports are not tree-shakable at all.

There is probably no optimal way (except from moving away from web pack to site), but the `index.module.js` of `@nextcloud/vue` is currently at 883K which seems to be a value that a lot of apps and also server easily exceed with the individual modules from dist.

Just out of curiosity, du you have the small test app in some repo? I'd be curious to see the actual impact in the bundles with `webpack-bundle-analyzer` or `vite-bundle-visualizer`

Reply:
> I think this is really bad for app that use more components and we should rather consider moving everything to importing the default export of the lib instead of the dist folder. As far as my tests went the dist imports are not tree-shakable at all.

I tried this for the forms app (just js files):
dist import: `9.694 MiB` (main branch)
default: `13.524 MiB` (`feat/move-to-module-import` branch)

I think for dist import only the really required parts are included while importing from default export might include some parts which are not flagged side effects free, like the emoji mart is always included (about 1MiB), e.g. compare:
<details>
`dist` import:

![Screenshot_20230208_124610](https://user-images.githubusercontent.com/1855448/217521758-63ed1e15-4537-428f-b2e9-ccbf16ba8d0a.png)

default import:

![Screenshot_20230208_124503](https://user-images.githubusercontent.com/1855448/217521782-0380c13f-95d6-41b2-bbd4-d0575e5adcfd.png)

</details>

> Just out of curiosity, du you have the small test app in some repo? I'd be curious to see the actual impact in the bundles with `webpack-bundle-analyzer` or `vite-bundle-visualizer`

No but it is not hard to create, see below:
<details>

Simply create a new project, install `@nextcloud/password-confirmation` (use this branch) and `@nextcloud/webpack-vue-config`.
Create a file `src/main.js` with this content:
```js
import { confirmPassword } from '@nextcloud/password-confirmation'
import '@nextcloud/password-confirmation/dist/style.css' // Required for dialog styles

export const foo = async () => {
    try {
        await confirmPassword()
        // Your logic
    } catch (error) {
        // Your error handling logic
    }
}
```
and `webpack.js` with
```js
const path = require('path')
const webpackConfig = require('@nextcloud/webpack-vue-config')

module.exports = webpackConfig
```

</details>

Reply:
Thanks for the insights. I reverted the import changes for now and just added the packages as external.

Switching the imports would still be good for larger apps that include more components anyways, but lets's keep this separate for now.

Reply:
I think this will work better when the vue components are provided as a esm package. Then even webpack can treeshake it more efficient.

> Switching the imports would still be good for larger apps

yes for all apps that e.g. include the emoji mart (one of the biggest dependencies), like the text app or the collectives app, this will be much better.